lms/djangoapps/teams/views.py

```diff
@@ -485,8 +487,20 @@ def has_object_permission(self, request, view, obj):
             """Returns true if the user is enrolled or is staff."""
             return has_team_api_access(request.user, obj.course_id)
 
+    class IsStaffOrTAOrReadOnly(IsStaffOrReadOnly):
+        """Permission that checks to see if the user is global staff, course
+        staff, or has discussion privileges. If none of those conditions are
+        met, only read access will be granted.
+        """
+
+        def has_object_permission(self, request, view, obj):
+            return (
+                has_discussion_privileges(request.user, obj.course_id) or
+                IsStaffOrReadOnly.has_object_permission(self, request, view, obj)
```

Comment:
@efischer19, the issue is that it is a nested class. If you use the syntax

super(TeamsDetailView.IsStaffOrPrivilegedOrReadOnly, self).has_object_permission(request, view, obj)

then it works (and all unit tests pass). That being said, I no longer feel strongly about which way you call the method-- from what I've read, super is really useful in the case of multiple inheritance, which is not in play here. I'm curious to hear what a Python expert (like @nedbat) would say about which way is preferable.